assert/assertions_test.go

```diff
@@ -730,10 +730,18 @@ func TestCondition(t *testing.T) {
 
 func TestDidPanic(t *testing.T) {
 
-	if funcDidPanic, _ := didPanic(func() {
-		panic("Panic!")
-	}); !funcDidPanic {
-		t.Error("didPanic should return true")
+	const panicMsg = "Panic!"
+
+	if funcDidPanic, msg := didPanic(func() {
```

Comment:
Would it be a bit clearer if we separated these `if` statements into two lines, assignment and `if` check?

e.g.
```
funcDidPanic, msg := didPanic(func() {...})
if !funcDidPanic || msg != panicMsg {
    ...
}
```

Reply:
I think it's fine as it is. Reads as naturally as any other go code to me.